Documentation/proposals/202304-agent-deployment-pattern-sidecar.md

```diff
@@ -31,6 +38,77 @@ With a visualization:
 
 ![Sidecar Deployment Pattern](assets/agent-deployment-pattern-sidecar/sidecar-deployment-pattern.png "Sidecar Deployment Pattern")
 
+## Implementation
+
+Following what other operators, e.g. OtelCollector and Istio, are doing already, we'll use [Admission Webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/) to mutate pods with the Prometheus Agent sidecar.
+
+The Prometheus-Operator project has implemented an admission webhook already. All that is left to do is add another route for pod mutation.
+
+In `pkg/admission/admission.go`, we'll do the following changes:
+
+*code redacted for improved reaadability*
```

Comment:
```suggestion
*code edited for improved readability*
```

Reply:
Tbh I don't know if you really need to show code here. The description is clear enough to me.

Reply:
I wanted to add it to the doc because we're not really following the conventional way when handling admission webhooks. We have a container that runs separated from the operator.

I was just thinking that providing a code snippet for whoever will implement this might help :)